linkis-computation-governance/linkis-client/linkis-cli/linkis-cli-application/src/main/java/com/webank/wedatasphere/linkis/cli/application/driver/UjesClientDriver.java

```diff
@@ -0,0 +1,595 @@
+ * @program: linkis-cli
+ * @description: UjesClientDriver communicates with UjesClient directly, calling apis to submit job/ query info etc, and translating UjesClient's exception.
+ * @author: shangda
+ * @create: 2021/02/22 21:25
+ */
+public class UjesClientDriver implements LinkisClientDriver {
+    private Logger logger = LoggerFactory.getLogger(UjesClientDriver.class);
+
+    private UJESClient client;
+    private UjesClientDriverContext context;
+
+    @Override
+    public void initDriver(LinkisClientDriverContext context) throws LinkisClientRuntimeException {
+        if (null == context) {
+            throw new ExecutorException(JobStatus.UNSUBMITTED, "EXE0008", ErrorLevel.ERROR, CommonErrMsg.ExecutionInitErr, "Cannot init UjesClientDriver: driver context is null");
+        }
+        if (!(context instanceof UjesClientDriverContext)) {
+            throw new ExecutorException(JobStatus.UNSUBMITTED, "EXE0009", ErrorLevel.ERROR, CommonErrMsg.ExecutionInitErr, "Cannot init UjesClientDriver: driver context is not instance of UjesClientDriverContext");
+        }
+
+        UjesClientDriverContext ujesContext = (UjesClientDriverContext) context;
+        this.context = ujesContext;
+
+        if (null == client) {
+            try {
+
+                AuthenticationStrategy authenticationStrategy;
+                if (StringUtils.isBlank(ujesContext.getAuthenticationStrategyStr()) ||
+                        !LinkisConstants.AUTH_STRATEGY_TOKEN.equalsIgnoreCase(ujesContext.getAuthenticationStrategyStr())) {
+                    authenticationStrategy = new StaticAuthenticationStrategy(); //this has to be newed here otherwise log-in fails for static
+                } else {
+                    authenticationStrategy = new TokenAuthenticationStrategy();
+                }
+
+                DWSClientConfigBuilder builder = DWSClientConfigBuilder.newBuilder();
+                DWSClientConfig config = ((DWSClientConfigBuilder) (
+                        builder.addServerUrl(ujesContext.getGatewayUrl())  //指定ServerUrl，linkis服务器端网关的地址,如http://{ip}:{port}
+                                .connectionTimeout(30000)   //connectionTimeOut 客户端连接超时时间
+                                .discoveryEnabled(false).discoveryFrequency(1, TimeUnit.MINUTES)  //是否启用注册发现，如果启用，会自动发现新启动的Gateway
+                                .loadbalancerEnabled(true)  // 是否启用负载均衡，如果不启用注册发现，则负载均衡没有意义
+                                .maxConnectionSize(5)   //指定最大连接数，即最大并发数
+                                .retryEnabled(false).readTimeout(ujesContext.getReadTimeoutMills())   //执行失败，是否允许重试
+                                .setAuthenticationStrategy(authenticationStrategy)   //AuthenticationStrategy Linkis认证方式
+                                .setAuthTokenKey(ujesContext.getTokenKey()).setAuthTokenValue(ujesContext.getTokenValue())))  //认证key，一般为用户名;  认证value，一般为用户名对应的密码
+                        .setDWSVersion(ujesContext.getDwsVersion()).build();  //linkis后台协议的版本，当前版本为v1
+
```

Comment:
Please translate comments into English. And you can add Chinese comments additionally in complex parts.